pkg/common/cns-lib/volume/manager.go

```diff
@@ -973,9 +973,21 @@ func (m *defaultManager) DetachVolume(ctx context.Context, vm *cnsvsphere.Virtua
 		volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
 		if volumeOperationRes.Fault != nil {
 			faultType = ExtractFaultTypeFromVolumeResponseResult(ctx, volumeOperationRes)
+
+			_, isManagedObjectNotFoundFault := volumeOperationRes.Fault.Fault.(*vim25types.ManagedObjectNotFound)
+			if isManagedObjectNotFoundFault {
+				// Detach failed with managed object not found, marking detach as
+				// successful, as Node VM is deleted and not present in the vCenter
+				// inventory.
+				log.Infof("DetachVolume: Node VM: %v not found on vCenter. Marking Detach for volume:%q successful.",
```

Comment:
Can you confirm if the managed object is the vm or the volume

Reply:
Managed object would be VM.

Reply:
where are we making sure ManagedObjectNotFound is for VM or Volume?

Reply:
can we utilize and confirm ManagedObjectNotFound is for VM we are trying to detach volume from?

```
// IsManagedObjectNotFound checks if err is the ManagedObjectNotFound fault.
// Returns true, if 'err' is a MnagedObjectNotFound fault for the intended
// 'moRef' object. Otherwise, return false.
func IsManagedObjectNotFound(err error, moRef types.ManagedObjectReference) bool {
	if soap.IsSoapFault(err) {
		fault, isNotFoundError := soap.ToSoapFault(err).VimFault().(types.ManagedObjectNotFound)
		return isNotFoundError && fault.Obj.Type == moRef.Type && fault.Obj.Value == moRef.Value
	}
	return false
}
```

Reply:
Made changes to make sure that ManagedObject is VM. Couldn't utilise this function since we don't have error with us.